platform-sdk/docs/core/signed-state-use.md

```diff
@@ -0,0 +1,38 @@
+# Signed State Use
+
+When using a `SignedState`, it is critically important to do so in a way that is thread safe and that does not cause
+improper use of the state's reference count. Failure to use a state object correctly is almost always fatal to a node.
+
+## Rules
+
+1. If an object of type `SignedState` is passed into a method, it is never thread safe to take a reference to the state
+   and use it after the method is returned.
+   a. A `SignedState` passed directly into a method is only guaranteed to be valid until the method returns.
+   b. If a `SignedState` is passed into a method, and there is need for the signed state to be used after the method
+      returns, then a new reservation on the signed state must be taken via
+      `ReservedSignedState ss = SignedState.reserve("reason")`.
```

Comment:
> what happens when an exception is thrown up the stack from where a ReserveSignedState is created and it has not been closed?

Crashing and not releasing a reservation on a state will almost always be fatal to the system. So we should do our best to minimize the amount of code that, if it crashes, is capable of leaking one of these reservations. Doing work in a try-with-resources block is a pretty safe way to do work, since if there is a crash anywhere in the block, we are guaranteed that the reservations will be freed.